src/NuGet.Services.EndToEnd/PushTests.cs

```diff
@@ -55,5 +55,32 @@ public PushTests(PushedPackagesFixture pushedPackages, ITestOutputHelper logger)
 
             await _clients.V3Search.WaitForPackageAsync(package.Id, package.Version, _logger);
         }
+
+        public async Task NewSemVer2PackageIsFilteredButAvailableInV3()
+        {
+            // Arrange
+            var package = await _pushedPackages.PushAsync(PackageType.SemVer2Prerelease, _logger);
+
+            // Wait for package to become available
+            await _clients.FlatContainer.WaitForPackageAsync(package.Id, package.Version, _logger);
+            // wait for package to be in both registrations
+            await _clients.Registration.WaitForPackageAsync(package.Id, package.Version, /*semver2*/ true, _logger);
+            await _clients.Registration.WaitForPackageAsync(package.Id, package.Version, /*semver2*/ false, _logger);
+            await _clients.V3Search.WaitForPackageAsync(package.Id, package.Version, _logger);
+
+            var shouldBeEmptyV3 = await _clients.V3Search.Query($"q={package.Id}");
+            var shouldBeEmptyV2 = await _clients.V3Search.SearchQuery($"q={package.Id}");
+            var shouldBeEmptyAutocomplete = await _clients.V3Search.Autocomplete($"q={package.Id}");
+            Assert.Equal(0, shouldBeEmptyV3.Data.Count);
+            Assert.Equal(0, shouldBeEmptyV2.Data.Count);
+            Assert.Equal(0, shouldBeEmptyAutocomplete.Data.Count);
+
+            var shouldNotBeEmptyV3 = await _clients.V3Search.Query($"q={package.Id}&semVerLevel=2.0.0");
```

Comment:
No need to test `/search/query` in my opinion. This is really an implementation detail and not a publicly used endpoint. You can push back if you want though.

Use `packageid:` for exact match.

`Assert.Equal(1, ...)` since we know the ID is unique.

Shouldn't `prerelease=true` be required?